<b>bedtools</b><br><br>
<b>A swiss-army knife of tools for genomic-analysis tasks.</b><br><br><b>Used to intersect, group, convert and count data in BAM, BED, GFF/GTF, VCF format.</b><br><br><b>Homepage: <a href="https://bedtools.readthedocs.io/en/latest/">home page</a></b><br><br>
- Intersect two files with respect to the sequences' strand and save the result to <i>path/to/output_file</i>:

<blockquote>bedtools intersect -a <i>path/to/file_1</i> -b <i>path/to/file_2</i> -s > <i>path/to/output_file</i></blockquote>
- Intersect two files with a left outer join, i.e. report each feature from <i>file_1</i> and NULL if no overlap with <i>file_2</i>:

<blockquote>bedtools intersect -a <i>path/to/file_1</i> -b <i>path/to/file_2</i> -lof > <i>path/to/output_file</i></blockquote>
- Using more efficient algorithm to intersect two pre-sorted files:

<blockquote>bedtools intersect -a <i>path/to/file_1</i> -b <i>path/to/file_2</i> -sorted > <i>path/to/output_file</i></blockquote>
- Group file <i>path/to/file</i> based on the first three and the fifth column and summarize the sixth column by summing it up:

<blockquote>bedtools groupby -i <i>path/to/file</i> -c 1-3,5 -g 6 -o sum</blockquote>
- Convert bam-formated file to a bed-formated one:

<blockquote>bedtools bamtobed -i <i>path/to/file</i">home page</a>bam > <i>path/to/file</i">home page</a>bed</blockquote>
- Find for all features in <i>file_1</i">home page</a>bed the closest one in <i>file_2</i">home page</a>bed and write their distance in an extra column (input files must be sorted):

<blockquote>bedtools closest -a <i>path/to/file_1</i">home page</a>bed -b <i>path/to/file_2</i">home page</a>bed -d</blockquote>
