<b>samtools</b><br><br>
<b>Tools for handling high-throughput sequencing (genomics) data.</b><br><br><b>Used for reading/writing/editing/indexing/viewing of data in SAM/BAM/CRAM format.</b><br><br>
- Convert a SAM input file to BAM stream and save to file:

<blockquote>samtools view -S -b <i>input.sam</i> > <i>output.bam</i></blockquote>
- Take input from stdin (-) and print the SAM header and any reads overlapping a specific region to stdout:

<blockquote><i>other_command</i> | samtools view -h - chromosome:start-end</blockquote>
- Sort file and save to BAM (the output format is automatically determined from the output file's extension):

<blockquote>samtools sort <i>input</i> -o <i>output.bam</i></blockquote>
- Index a sorted BAM file (creates <i>sorted_input.bam.bai</i>):

<blockquote>samtools index <i>sorted_input.bam</i></blockquote>
- Print alignment statistics about a file:

<blockquote>samtools flagstat <i>sorted_input</i></blockquote>
- Count alignments to each index (chromosome / contig):

<blockquote>samtools idxstats <i>sorted_indexed_input</i></blockquote>
- Merge multiple files:

<blockquote>samtools merge <i>output</i> <i>input1 input2 â€¦</i></blockquote>
- Split input file according to read groups:

<blockquote>samtools split <i>merged_input</i></blockquote>
